OpenRA.Mods.Cnc/ImportTiberianDawnLegacyMapCommand.cs

```diff
@@ -0,0 +1,209 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using OpenRA.FileFormats;
+using OpenRA.Mods.Common.UtilityCommands;
+using OpenRA.Primitives;
+
+namespace OpenRA.Mods.Cnc.UtilityCommands
+{
+	class ImportTiberianDawnLegacyMapCommand : IUtilityCommand
+	{
+		public string Name { get { return "--import-map"; } }
+
+		public bool ValidateArguments(string[] args)
+		{
+			return args.Length >= 2;
+		}
+
+		const int MapSize = 64; // NOTE: original engine limitation
+
+		Map map;
+		List<string> players = new List<string>();
+		MapPlayers mapPlayers;
+
+		[Desc("FILENAME", "Convert a legacy INI/MPR map to the OpenRA format.")]
+		public void Run(ModData modData, string[] args)
+		{
+			// HACK: The engine code assumes that Game.modData is set.
+			Game.ModData = modData;
+			Game.ModData.MountFiles();
+
+			var rules = Game.ModData.RulesetCache.Load();
+			var filename = args[1];
+
+			using (var stream = Game.ModData.ModFiles.Open(filename))
```

Comment:
`mono --debug OpenRA.Utility.exe ra --import-map /home/matthias/Downloads/scg01ea.ini` works just fine for me.